### PR TITLE
Remove FormProcessorCouch and Couch db accessors

### PR DESCRIPTION
**Pull request by millerdev**
Remove everything in `corehq.form_processor.backends.couch` as well as `should_use_sql_backend` and as many things as possible that were only used by that code.

:tropical_fish: Review by commit.

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Automated test coverage

No new tests were added. Unused features are being removed. Tests that depended on those features were removed previously or in this PR.

### QA Plan

No QA.

### Safety story

All domains in active environments are using the SQL backend, so everything removed in this PR, including code removed from tests, should be unused or unnecessary.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

File: corehq/ex-submodules/casexml/apps/case/dbaccessors/related.py [hunk 1]
```diff
@@ -1,84 +1,4 @@
...
-def _get_keys_for_reverse_index_view(domain, case_ids, relationship=None):
-    assert not isinstance(case_ids, (six.text_type, bytes))
-    if relationship is None:
-        return [[domain, case_id, 'reverse_index', reln]
-                for case_id in case_ids for reln in [CASE_INDEX_CHILD, CASE_INDEX_EXTENSION]]
-    else:
-        return [[domain, case_id, 'reverse_index', relationship] for case_id in case_ids]
 
 
 def get_reverse_indices_json(domain, case_id):
```
Comment (millerdev) [inline corehq/ex-submodules/casexml/apps/case/dbaccessors/related.py:4]: This and probably other things in this file will be removed in a separate PR.

**Review by dannyroberts — APPROVED** [review]

**Review by snopoke — APPROVED** [review]

**Comment by dannyroberts** [comment]
(Build failures look real)
